pkg/sources/postman/postman.go

```diff
@@ -0,0 +1,672 @@
+		s.client.HTTPClient = common.RetryableHTTPClientTimeout(3)
+	case *sourcespb.Postman_Unauthenticated:
+		s.client = nil
+		// No client needed if reading from local
+	default:
+		return errors.New("credential type not implemented for Postman")
+	}
+
+	return nil
+}
+
+// Chunks scans the Postman source and sends the data to the chunks chan.
+// It scans the local environment, collection, and workspace files, and then scans the Postman API if a token is provided.
+// The Postman source is different to our other sources in that we are not relying on the data we read from the source to contain
+// all the information we need to scan, i.e, a git chunk contains all the information needed to check if a secret is present in that chunk.
+// Postman on the other hand requires us to build context (the keywords and variables) as we scan the data.
+// Check out the postman UI to see what I mean.
+// Metadata is used to track information that informs the source of the chunk (e.g. the workspace -> collection -> request -> variable hierarchy).
+func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, _ ...sources.ChunkingTarget) error {
+	// Scan local environments
+	for _, envPath := range s.conn.EnvironmentPaths {
+		env := VariableData{}
+		contents, err := os.ReadFile(envPath)
+		if err != nil {
+			return err
+		}
+		if err = json.Unmarshal(contents, &env); err != nil {
+			return err
+		}
+		s.scanVariableData(ctx, chunksChan, Metadata{EnvironmentName: env.ID, fromLocal: true, Link: envPath}, env)
+	}
+
+	// Scan local workspaces
+	for _, collectionPath := range s.conn.CollectionPaths {
+		collection := Collection{}
+		contents, err := os.ReadFile(collectionPath)
+		if err != nil {
+			return err
+		}
+		if err = json.Unmarshal(contents, &collection); err != nil {
+			return err
+		}
+		s.scanCollection(ctx, chunksChan, Metadata{CollectionInfo: collection.Info, fromLocal: true, Link: collectionPath}, collection)
+	}
+
+	// Scan local workspaces
+	for _, workspacePath := range s.conn.WorkspacePaths {
+		// check if zip file
+		workspace := Workspace{}
+		if strings.HasSuffix(workspacePath, ".zip") {
+			r, err := zip.OpenReader(workspacePath)
+			if err != nil {
+				return err
+			}
+			for _, file := range r.File {
+				rc, err := file.Open()
+				if err != nil {
+					return err
+				}
+				contents, err := io.ReadAll(rc)
+				if err != nil {
+					return err
+				}
+				if strings.Contains(file.Name, "collection") {
+					// read in the collection then scan it
+					c := Collection{}
+					if err = json.Unmarshal(contents, &c); err != nil {
+						return err
+					}
+					workspace.CollectionsRaw = append(workspace.CollectionsRaw, c)
+				}
+				if strings.Contains(file.Name, "environment") {
+					e := VariableData{}
+					if err = json.Unmarshal(contents, &e); err != nil {
+						return err
+					}
+					workspace.EnvironmentsRaw = append(workspace.EnvironmentsRaw, e)
+				}
+			}
+			r.Close()
```

Comment:
The difficult thing about this approach is that we aren't closing `r` when we encounter an error that causes us to return from the function immediately. That's what `defer` solves, but unfortunately not within a loop body.

I'm okay with this working for the happy path for now, but a followup PR should fix this leak (which I'm happy to do too).